src/Sql/Sql/ManagedInstanceDnsAlias/Cmdlet/MoveSqlManagedInstanceDnsAlias.cs

```diff
@@ -0,0 +1,252 @@
+        /// </summary>
+        [Parameter(Mandatory = true, ValueFromPipeline = true, ParameterSetName = MoveByNameAndSourceResourceIdParameterSet, Position = 2, HelpMessage = "Resource ID of the source managed instance DNS alias.")]
+        [Parameter(Mandatory = true, ValueFromPipelineByPropertyName = true, ParameterSetName = MoveByParentObjectAndSourceResourceIdParameterSet, Position = 1, HelpMessage = "Resource ID of the source managed instance DNS alias.")]
+        [ValidateNotNullOrEmpty]
+        [Alias("Id")]
+        public string SourceResourceId { get; set; }
+
+        /// <summary>
+        /// Gets or sets whether or not to run this cmdlet in the background as a job
+        /// </summary>
+        [Parameter(Mandatory = false, HelpMessage = "Run cmdlet in the background")]
+        public SwitchParameter AsJob { get; set; }
+
+        protected override IEnumerable<AzureSqlManagedInstanceDnsAliasModel> GetEntity()
+        {
+            // Since we didn't construct source DNS alias resource Id, for this parameter set, in the ExecuteCmdlet method,
+            // we can do it now, since ModelAdapter is initialized at this point (it was initialized after entering the base.ExecuteCmdlet method).
+            if(ParameterSetName == MoveByNamesParameterSet)
+            {
+                ConstructSourceResourceId(SourceResourceGroupName, SourceInstanceName, SourceName, ModelAdapter.Context.Subscription.Id);
+            }
+            return ModelAdapter.ListManagedInstanceDnsAliases(this.ResourceGroupName, this.InstanceName);
+        }
+ 
+        private void ConstructSourceResourceId(string sourceResourceGroupName, string sourceManagedInstanceName, string sourceDnsAliasName, string subscriptionId)
+        {
+            SourceResourceId = string.Format(
+                    ManagedInstanceDnsAliasResourceIdTemplate,
+                    subscriptionId,
+                    sourceResourceGroupName,
+                    sourceManagedInstanceName,
+                    sourceDnsAliasName);
+        }
+
+        private void ConstructSourceResourceId(AzureSqlManagedInstanceModel sourceInstance, string sourceDnsAliasName, string subscriptionId)
+        {
+            SourceResourceId = string.Format(
+                    ManagedInstanceDnsAliasResourceIdTemplate,
+                    subscriptionId,
+                    sourceInstance.ResourceGroupName,
+                    sourceInstance.ManagedInstanceName,
+                    sourceDnsAliasName);
+        }
+
+        private void ConstructSourceResourceId(AzureSqlManagedInstanceDnsAliasModel sourceDnsAlias)
+        {
+            SourceResourceId = sourceDnsAlias.Id;
+            SourceName = sourceDnsAlias.DnsAliasName;
+        }
+
+        private void ParseParentObjectParameters()
+        {
+            ResourceGroupName = InstanceObject.ResourceGroupName;
+            InstanceName = InstanceObject.ManagedInstanceName;
```

Comment:
nitpick, these kind of 2liner voids with sideeffects are not a great practice IMO (when sideeffect is not required).
Also we're not really checking for the 'input' here, e.g. the latter one kinda prays we'll have `InstanceObject` when invoked which is not really great (think broader than this PR, i.e. copy pasting, refactoring, lack of tests etc) + I don't think you actually typed less code this way 😄

Reply:
Yeah, they should have some checks running at least. This Move command class had more parameter sets initially so I used this to avoid copy pasting for multiple parameter sets, but I ommited refactoring this when I removed some of the unnecessary parameter sets.
TLDR: I will inline those :)

Reply:
I've inlined those functions